netstandard/tools/Trimming/FileNode.cs

```diff
@@ -0,0 +1,153 @@
+                            foreach (var referenceCandidate in referenceCandidates)
+                            {
+                                if (allFiles.TryGetValue(referenceCandidate, out referencedFile))
+                                {
+                                    break;
+                                }
+                            }
+
+                            if (referencedFile != null)
+                            {
+                                dependencies.Add(referencedFile);
+                            }
+                            else
+                            {
+                                // static dependency that wasn't satisfied, this can happen if folks use 
+                                // lightup code to gaurd the static dependency.
+                                // this can also happen when referencing a package that isn't implemented
+                                // on this platform but don't fail the build here
+                            }
+                        }
+
+                        for (int i = 1, count = reader.GetTableRowCount(TableIndex.ModuleRef); i <= count; i++)
+                        {
+                            var moduleRef = reader.GetModuleReference(MetadataTokens.ModuleReferenceHandle(i));
+                            var moduleName = reader.GetString(moduleRef.Name);
+
+                            var moduleRefCandidates = new[] { moduleName, moduleName + ".dll", moduleName + ".so", moduleName + ".dylib" };
+
+                            FileNode referencedNativeFile = null;
+                            foreach (var moduleRefCandidate in moduleRefCandidates)
+                            {
+                                if (allFiles.TryGetValue(moduleRefCandidate, out referencedNativeFile))
+                                {
+                                    break;
+                                }
+                            }
+
+                            if (referencedNativeFile != null)
+                            {
+                                dependencies.Add(referencedNativeFile);
+                            }
+                            else
+                            {
+                                // DLLImport that wasn't satisfied
+                            }
+                        }
+                    }
+                }
+            }
+            catch(BadImageFormatException)
+            {
+                // not a PE
+            }
+
+            // allow for components to specify their dependencies themselves, by placing a file next to their source file.
+            var additionalDependenciesFile = SourceFile + AdditionalDependenciesFileSuffix;
+
+            if (File.Exists(additionalDependenciesFile))
+            {
+                foreach(var additionalDependency in File.ReadAllLines(additionalDependenciesFile))
+                {
+                    FileNode additionalDependencyFile;
```

Comment:
Do you want to add in a commenting mechanism? Something like line starts with `#` I guess it doesn't really matter given that you are only doing a dictionary look-up with the line.